Upgrade operator-sdk to v1.17.0

kevinearls opened the pull request:
Signed-off-by: Kevin Earls <kearls@redhat.com>

## Which problem is this PR solving?
- We would like to upgrade to operator =-sdk to v1,17 or newer
- 
## Short description of the changes
- Upgrades to operator-sdk according to https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/

kevinearls committed:
First step - upgrade to v1.14.0

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
First step - upgrade to v1.14.0

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Merge branch 'main' into update-operator-sdk

kevinearls committed:
Upgrade to operator-sdk v1.17.0

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Upgrade to operator-sdk v1.17.0

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Update to go version 1.17

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Update to go version 1.17

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Make sure go 1.17 is used everywhere

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Make sure go 1.17 is used everywhere

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls committed:
Merge branch 'jaegertracing:main' into update-operator-sdk

kevinearls commented:
@pavolloffay  @rubenvp8510  @esnible  Please review

Commit:
Improve reliability of some scripts to download third party tools (#1779)

* Improve reliability of some scripts to download third party tools #1778 #1750

Signed-off-by: Israel Blancas <iblancas@redhat.com>

* Fix finx kind binary

Signed-off-by: Israel Blancas <iblancas@redhat.com>

frzifus committed:
update release.md (#1819)

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.7.0 to 1.7.1 (#1818)

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.0 to 1.7.1.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.0...v1.7.1)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Pavol Loffay <p.loffay@gmail.com>

pavolloffay reviewed (commented):
Missing parts:
* remove support for admissionReviewVersions v1beta1
* change the golang version to 1.17 in all CI workflows and all dockerfiles 
* `--ignore-not-found` is missing in the Makefile

rubenvp8510 commented on bundle/manifests/jaeger-operator.clusterserviceversion.yaml:
```diff
@@ -23,7 +23,7 @@ metadata:
       distributed systems
     operators.openshift.io/infrastructure-features: '["disconnected"]'
-    operators.operatorframework.io/builder: operator-sdk-v1.13.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.16.0+git
```
should this be 1.16 or 1.17? 🤔

kevinearls replied:
@rubenvp8510 I'm not sure.  This code is all generated, right?

rubenvp8510 commented on bundle/manifests/jaeger-operator.clusterserviceversion.yaml:
```diff
@@ -388,14 +388,20 @@ spec:
                 - --secure-listen-address=0.0.0.0:8443
                 - --upstream=http://127.0.0.1:8383/
                 - --logtostderr=true
-                - --v=10
```
why this change?

kevinearls replied:
Again, I don't know, this is generated code.

dependabot[bot] committed:
Bump actions/setup-go from 2.2.0 to 3 (#1816)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 2.2.0 to 3.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v2.2.0...v3)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rubenvp8510 commented on .github/workflows/base-checks.yaml:
```diff
@@ -17,7 +17,7 @@ jobs:
       uses: actions/setup-go@v2.2.0
       with:
-        go-version: 1.16
+        go-version: 1.17
```
Is not required, but I think is a good opportunity to upgrade golang to a more recent version.

kevinearls replied:
@frzifus Actually update to 1.17 is requred: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/

It should be in the go.mod file

frzifus replied:
Now iam confused. you are right, your go.mod file says [1.17](https://github.com/jaegertracing/jaeger-operator/pull/1813/files/17ac0d391afce6bb80f4a523a14dd228f6be5970..0fc655266338b959893753738c32916f0d35eec5#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R3). But github doesnt show a diff to the main branch. There it says [1.16](https://github.com/jaegertracing/jaeger-operator/blob/main/go.mod#L3)?

frzifus replied:
oh damn its actually a diff between earlier commits on your branch. :facepalm: ... here is the change to [1.17](https://github.com/jaegertracing/jaeger-operator/pull/1813/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6) ...

kevinearls committed:
Trying to rebase

Signed-off-by: Kevin Earls <kearls@redhat.com>

kevinearls commented:
@pavolloffay  yes, I had already looked for "admissionReviewVersions" and not found it, which is why I asked.

kevinearls committed:
Added  --ignore-not-found for undeploy

Signed-off-by: Kevin Earls <kearls@redhat.com>

frzifus commented:
Not sure if we can merge it without fixing the `DCO Action required after 1s — DCO` thing.

I tried to realign things on https://github.com/frzifus/jaeger-operator/commits/update-operator-sdk
If you like, you can switch to that branch by:
```
git checkout main # switching to main branch
git branch -D update-operator-sdk # removing your local feature branch
git remote add frzifus git@github.com:frzifus/jaeger-operator.git # adding frzifus as source
git fetch frzifus update-operator-sdk # getting the changes
git checkout update-operator-sdk # checkout the new branch
git push origin update-operator-sdk --force # note: make sure origin points to your remote repo
```

to verify if `origin` points to your remote repo, use `git remove -v`

kevinearls commented:
@pavolloffay @rubenvp8510 Can you review please and merge if it looks ok.  I think I have addressed all of your comments.  

@frzifus and I spent some time looking at the DCO failure and haven't yet figured out how to fix it but I don't think that should be a blocker.
